paz/models/classification/cnn2Plus1.py

```diff
@@ -0,0 +1,354 @@
+    x = BatchNormalization()(x)
+    x = ReLU()(x)
+    x = ResizeVideo(height // 2, width // 2)(x)
+
+    # Block 1
+    x = add_residual_block(x, 16, (3, 3, 3))
+    x = ResizeVideo(height // 4, width // 4)(x)
+
+    # Block 2
+    x = add_residual_block(x, 32, (3, 3, 3))
+    x = ResizeVideo(height // 8, width // 8)(x)
+
+    # Block 3
+    x = add_residual_block(x, 64, (3, 3, 3))
+    return x
+
+
+def original_18(input_layer, height, width):
+    """ Architecture is a CNN2+1D version based on the original paper.
+        # Arguments
+            input_layer: Tensorflow input layer of the network
+            height: Height of the input video
+            width: Width of the input video
+    """
+    x = Conv2Plus1D(filters=16, kernel_size=(3, 7, 7), padding='same')(input_layer)
+    x = BatchNormalization()(x)
+    x = ReLU()(x)
+    x = ResizeVideo(height // 2, width // 2)(x)
+
+    # Block 1
+    x = add_residual_block(x, 64, (3, 3, 3))
+    x = add_residual_block(x, 64, (3, 3, 3))
+    x = ResizeVideo(height // 4, width // 4)(x)
+
+    # Block 2
+    x = add_residual_block(x, 128, (3, 3, 3))
+    x = add_residual_block(x, 128, (3, 3, 3))
+    x = ResizeVideo(height // 8, width // 8)(x)
+
+    # Block 3
+    x = add_residual_block(x, 256, (3, 3, 3))
+    x = add_residual_block(x, 256, (3, 3, 3))
+    x = ResizeVideo(height // 16, width // 16)(x)
+
+    # Block 4
+    x = add_residual_block(x, 512, (3, 3, 3))
+    x = add_residual_block(x, 512, (3, 3, 3))
+    return x
+
+
+def CNN2Plus1D(weights=None, input_shape=(38, 96, 96, 3), seed=305865,
+               architecture: Architecture_Options = 'CNN2Plus1D',
+               tmp_weights_path=None):
+    """Binary Classification for videos with 2+1D CNNs.
+    # Arguments
+        weights: ``None`` or string with pre-trained dataset. Valid datasets
+            include only ``VVAD-LRS3``.
+        input_shape: List of integers. Input shape to the model in following format: (frames, height, width, channels)
+        e.g. (38, 96, 96, 3).
+        seed: Integer. Seed for random number generator.
+        architecture: String. Name of the architecture to use. Currently supported: 'CNN2Plus1D', 'CNN2Plus1D_Filters',
+            'CNN2Plus1D_Layers', 'CNN2Plus1D_Light'. 'CNN2Plus1D_18' is only available without weights.
+
+    # Reference
+        - [A Closer Look at Spatiotemporal Convolutions for Action Recognition](https://arxiv.org/abs/1711.11248v3)
+        - [Video classification with a 3D convolutional neural network]
+        (https://www.tensorflow.org/tutorials/video/video_classification#load_and_preprocess_video_data)
+    """
+    if len(input_shape) != 4:
+        raise ValueError(
+            '`input_shape` must be a tuple of 4 integers. '
+            'Received: %s' % (input_shape,))
+    options = get_args(Architecture_Options)
+    assert architecture in options, f"'{architecture}' is not in {options}"
+
+    random.seed(seed)
```

Comment:
I don't know if this is really the right place to put this. If possible I would delete it.